clients/extension/scripts/extension-entry.js

```diff
@@ -5,133 +5,46 @@
  */
 'use strict';
 
-const lighthouse = require('../../../lighthouse-core/index.js');
-const Config = require('../../../lighthouse-core/config/config.js');
-const defaultConfig = require('../../../lighthouse-core/config/default-config.js');
-const i18n = require('../../../lighthouse-core/lib/i18n/i18n.js');
-
-const ExtensionProtocol = require('../../../lighthouse-core/gather/connections/extension.js');
-const log = require('lighthouse-logger');
+const defaultCategories = [{
+  id: 'performance',
+  title: 'Performance',
+}, {
+  id: 'accessibility',
+  title: 'Accessibility',
+}, {
+  id: 'best-practices',
+  title: 'Best Practices',
+}, {
+  id: 'seo',
+  title: 'SEO',
+}, {
+  id: 'pwa',
+  title: 'Progressive Web App',
+}];
 
 /** @typedef {import('../../../lighthouse-core/gather/connections/connection.js')} Connection */
+/** @typedef {{selectedCategories: string[], device: string}} Settings */
```

Comment:
enums would be useful, if possible

Reply:
b/c device is grabbed from the dom, and must be coerced, making this an enum just seems like typescript-for-typescript's-sake

```
diff --git a/clients/extension/scripts/extension-controller.js b/clients/extension/scripts/extension-controller.js
index 7cc8365de..feb703b4d 100644
--- a/clients/extension/scripts/extension-controller.js
+++ b/clients/extension/scripts/extension-controller.js
@@ -23,7 +23,7 @@ const defaultCategories = [{
   title: 'Progressive Web App',
 }];
 
-/** @typedef {{selectedCategories: string[], device: string}} Settings */
+/** @typedef {{selectedCategories: string[], device: 'mobile' | 'desktop'}} Settings */
 
 const STORAGE_KEY = 'lighthouse_audits';
 const SETTINGS_KEY = 'lighthouse_settings';
diff --git a/clients/extension/scripts/popup.js b/clients/extension/scripts/popup.js
index 655d0c225..04d0c8500 100644
--- a/clients/extension/scripts/popup.js
+++ b/clients/extension/scripts/popup.js
@@ -165,7 +165,8 @@ async function initPopup() {
     const checkboxes = /** @type {NodeListOf<HTMLInputElement>} */
       (optionsEl.querySelectorAll('.options__categories input:checked'));
     const selectedCategories = Array.from(checkboxes).map(input => input.value);
-    const device = /** @type {HTMLInputElement} */ (find('input[name="device"]:checked')).value;
+    const deviceInputEl = /** @type {HTMLInputElement} */ (find('input[name="device"]:checked'));
+    const device = /** @type {'mobile' | 'desktop'} */ (deviceInputEl.value);
 
     ExtensionController.saveSettings({
       selectedCategories,

```